lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.ts

```diff
@@ -143,37 +144,45 @@ export class AttachFileCloudWidgetComponent extends UploadCloudWidgetComponent i
-                destinationFolderPath = this.getAliasAndRelativePathFromDestinationFolderPath(this.field.params.fileSource.destinationFolderPath.value);
-                destinationFolderPath.path = this.replaceAppNameAliasWithValue(destinationFolderPath.path);
-            }
-
-            if (this.isPathVariableType(DestinationFolderPathType.FOLDER_TYPE)) {
-                rootNodeId = this.field.params.fileSource.destinationFolderPath.value;
-            }
-        }
 
-        if (!rootNodeId) {
-            try {
-                const nodeId = await this.getNodeIdBasedOnPath(destinationFolderPath);
-                rootNodeId = nodeId ? nodeId : destinationFolderPath.alias;
-            } catch (error) {
-                this.logService.error(error);
-            }
+        switch (this.field?.params?.fileSource?.destinationFolderPath?.type) {
+            case DestinationFolderPathType.STATIC_TYPE:
+                rootNodeId = await this.getNodeIdFromPath(this.field.params.fileSource.destinationFolderPath);
+                break;
+            case DestinationFolderPathType.STRING_TYPE:
+                rootNodeId = await this.getNodeIdFromPath(this.field.params.fileSource.destinationFolderPath);
+                break;
+            case DestinationFolderPathType.FOLDER_TYPE:
+                rootNodeId = await this.getNodeIdFromFolderVariableValue(this.field.params.fileSource.destinationFolderPath);
+                break;
+            default:
+                rootNodeId = await this.getNodeIdFromPath({ type: '', value: AttachFileCloudWidgetComponent.ALIAS_USER_FOLDER });
+                break;
         }
 
         return rootNodeId;
     }
 
-    private async getNodeIdBasedOnPath(destinationFolderPath: DestinationFolderPathModel) {
+   async getNodeIdFromPath(destinationFolderPath: DestinationFolderPath): Promise<string> {
         let nodeId: string;
-        if (destinationFolderPath.path) {
-            nodeId = await this.contentNodeSelectorService.fetchNodeIdFromRelativePath(destinationFolderPath.alias, { relativePath: destinationFolderPath.path });
+        let destinationPath: DestinationFolderPathModel;
+        destinationPath =  this.getAliasAndRelativePathFromDestinationFolderPath(destinationFolderPath.value);
+        destinationPath.path = this.replaceAppNameAliasWithValue(destinationPath.path);
+        try {
+            nodeId = await this.contentNodeSelectorService.getNodeIdFromPath(destinationPath);
+        } catch (error) {
+            this.logService.error(error);
         }
-        if (!nodeId) {
-            nodeId = await this.contentNodeSelectorService.fetchAliasNodeId(destinationFolderPath.alias);
+
+        return nodeId;
+    }
+
+    async getNodeIdFromFolderVariableValue(destinationFolderPath: DestinationFolderPath): Promise<string> {
+        let nodeId: string;
+        try {
+            nodeId = await this.contentNodeSelectorService.verifyAndReturnNodeId(destinationFolderPath.value);
```

Comment:
not sure about this verifyAndReturnNodeId() function.
The AND word indicates that it does more than one things.
```
const isExistingFolder = await this.contentNodeSelectorService.isExistingNode(destinationFolderPath.value);

const nodeId = isExistingFolder ? folderId : this.getNodeId(ContentCloudNodeSelectorService.ALIAS_USER_FOLDER).toPromise();
```

Something like that is more clear to me. 
or isNodeAvailable() instead of isExistingNode(), as you wish.
Because it is more clear to me in this way it doesn't mean it is more clear for everybody 😛 Please consider if what I am saying makes sense and if you find it good, let's change it 😀